### PR TITLE
adding ruby test

### PR DESCRIPTION
**Pull request by marjanfariborz**
Change-Id: I8b2417cbb1e879328e2e28c3e62133162f42a062

#69

### CODE REVIEW

File: configs/common/FileSystemConfig.py [hunk 1]
```diff
@@ -59,6 +59,19 @@ def config_filesystem(options):
     # Set up /proc
     procdir = joinpath(fsdir, 'proc')
     mkdir(procdir)
+    cpu_clock = '0'
```
Comment (powerjg) [inline configs/common/FileSystemConfig.py]: Is this change required for your test or something from upstream?

File: configs/common/FileSystemConfig.py [hunk 2]
```diff
@@ -67,8 +80,6 @@ def config_filesystem(options):
                   'model           : 0\n'                         + \
                   'model name      : Generic\n'                   + \
                   'stepping        : 0\n'                         + \
-                  'cpu MHz         : %0.3d\n'                       \
```
Comment (powerjg) [inline configs/common/FileSystemConfig.py]: Ditto.

File: configs/common/Options.py [hunk 3]
```diff
@@ -45,12 +45,12 @@
 from m5.defines import buildEnv
 from m5.objects import *
 
-from .Benchmarks import *
-from . import CpuConfig
-from . import BPConfig
-from . import HWPConfig
-from . import MemConfig
-from . import PlatformConfig
+from common.Benchmarks import *
```
Comment (powerjg) [inline configs/common/Options.py]: Ditto.

File: tests/gem5/hello_Ruby/test_ruby.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2017 Mark D. Hill and David A. Wood
```
Comment (powerjg) [inline tests/gem5/hello_Ruby/test_ruby.py]: Copyright should be Regents of University of California. Check one of the other files we've added recently or there should be this info in darchr/docs, too.

File: tests/gem5/hello_Ruby/test_ruby.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# Authors: Sean Wilson
```
Comment (powerjg) [inline tests/gem5/hello_Ruby/test_ruby.py]: Your name here ;)

File: tests/gem5/hello_Ruby/test_ruby.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+}
+isa='x86'
+for binary in test_progs:
+    import os
+    ref_path = joinpath(getcwd(), 'ref')
+    gem5_verify_config(
+        name='test_ruby'+binary,
+        verifiers=(),
+        fixtures=(),
+        config=joinpath(config.base_dir,'configs','example',
```
Comment (powerjg) [inline tests/gem5/hello_Ruby/test_ruby.py]: If you're using the random tester, I don't think you need the `test_progs`.

**Review by powerjg — COMMENTED** [review]
Thanks, Marjan! I'm excited to have these tests!

When using git, I suggest that when you're adding things you always explicitly add files (i.e., don't use `git add .`). Please remove the binary files. It may help if you run `git status` before committing as well.

To remove the files, you can unstage them then recommit (e.g, `git commit --amend`) and force push (`git push -f`).

**Comment by marjanfariborz** [comment]
Sorry about that. I'll fix it right now.

Thanks

File: tests/gem5/hello_Ruby/test_ruby.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# Authors: Marjan Fariborz
+
+'''
+Test file for the util m5 exit assembly instruction.
```
Comment (powerjg) [inline tests/gem5/hello_Ruby/test_ruby.py]: The description needs to be updated.

File: tests/gem5/hello_Ruby/test_ruby.py [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+'''
+from testlib import *
+
+#test_progs = {
```
Comment (powerjg) [inline tests/gem5/hello_Ruby/test_ruby.py]: Also, clean up and remove any comments that aren't needed.

File: tests/gem5/hello_Ruby/test_ruby.py [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+#    'hello64-static', 'hello64-dynamic', 'hello32-static',
+#}
+isa='x86'
+#for binary in test_progs:
+import os
```
Comment (powerjg) [inline tests/gem5/hello_Ruby/test_ruby.py]: used?

File: tests/gem5/hello_Ruby/test_ruby.py [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+'''
+Test file for the util m5 exit assembly instruction.
+'''
+from testlib import *
+
+#test_progs = {
+#    'hello64-static', 'hello64-dynamic', 'hello32-static',
+#}
+isa='x86'
```
Comment (powerjg) [inline tests/gem5/hello_Ruby/test_ruby.py]: We might want to use "NULL" here, if possible. Also, we should specify which protocols to test.

File: tests/gem5/hello_Ruby/test_ruby.py [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+#for binary in test_progs:
+import os
+ref_path = joinpath(getcwd(), 'ref')
+gem5_verify_config(
+    name='test_ruby',
+    verifiers=(),
+    fixtures=(),
+    config=joinpath(config.base_dir,'configs','example',
+    'ruby_random_test.py'),
+    config_args=[],
```
Comment (powerjg) [inline tests/gem5/hello_Ruby/test_ruby.py]: Probably want to specify some args here, like the number of cores and the number of loads.

**Review by powerjg — COMMENTED** [review]
I'm testing this real fast. :) I might make some updates directly myself, FYI.

**Comment by powerjg** [comment]
I added a patch to fixup the things I mentioned in the review. It looks like it's working!

Now, all that's left is to rebase on master (or upstream since it's going there) and to update the commit message (e.g., "tests:....")
